File: ios/Employee.swift
//
//  Employee.swift
//  Eventlib
//
//  Created by Alexandre carvalho Marques on 27/06/22.
//  Copyright © 2022 Facebook. All rights reserved.
//

import UIKit

public class Employee {
    var name: String
    var age: Int
    var designation: String
    var salary: Double
    
    init(name: String){
        self.name = name
    }

    func getName() -> String {
       return name
     }

     /* Print the Employee details */
     func printEmployee() -> (){
        print("Name: \(self.name)")
        print("Age: \(self.age)")
        print("Designation: \(self.designation)")
        print("Salary: \(self.salary)")
     }
}

File: android/src/main/java/com/reactnativeeventlibrary/Employee.java
package com.reactnativeeventlibrary;
import android.util.Log;
import java.io.*;

public class Employee {
  String name;
  int age;
  String designation;
  double salary;
  private static final String TAG = "MyEvent";

  // This is the constructor of the class Employee
  public Employee(String name) {
    this.name = name;
  }

  public String getName() {
    return name;
  }

  public void setAge(int age) {
    age = age;
  }

  public void setDesignation(String desig) {
    designation = desig;
  }

  public void setSalary(double salary) {
    salary = salary;
  }

  /* Print the Employee details */
  public void printEmployee() {
    Log.i(TAG,"Name:" + name);
    Log.i(TAG, "Age:" + age );
    Log.i(TAG, "Designation:" + designation );
    Log.i(TAG, "Salary:" + salary);
  }
}
